client/state/data-layer/wpcom-http/freshness-verifier.js

```diff
@@ -0,0 +1,69 @@
+export const CLEAN_OLDER_THAN = 3 * 60 * 1000;
+
+/** @type {Date} holds timestamp of the last time a cleaning operation was performed */
+let lastCleaningTime;
+
+/** @type {Map} holds request action keys, and last execution times */
+const history = new Map();
+
+/**
+ * Generate a deterministic key for comparing request actions
+ *
+ * @param   {Object} action redux action
+ * @returns {String}        action key
+ */
+const buildActionKey = ( action ) => JSON.stringify(
+	action.meta ? { ...action, meta: undefined } : action
+);
```

Comment:
Hi @dmsnell, Yes it is true that freshness is supposed to be used with gets, and gets can be defined by the path and query parameters, but on dispatchRequest we don't know yet how the action is mapped to path and query parameters. On the dedupe request we already have access to this mapping so the key build there can take advantage of that.